File: frontend/src/types/broker.ts
export type BrokerStatus = {
  adapter: string;
  mode: string;
  connected: boolean;
  details: string;
};

export type Order = {
  id: number;
  symbol: string;
  side: string;
  order_type: string;
  quantity: number;
  limit_price?: number | null;
  status: string;
  mode: string;
  submitted_at?: string | null;
};

export type Execution = {
  id: number;
  order_id: number;
  executed_at: string;
  price: number;
  quantity: number;
  fees?: number | null;
};

File: backend/app/schemas/broker.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class BrokerStatusResponse(BaseModel):
    adapter: str
    mode: str
    connected: bool
    details: str


class OrderCreateRequest(BaseModel):
    symbol: str
    side: str
    quantity: float
    order_type: str = "market"
    limit_price: float | None = None


class OrderResponse(BaseModel):
    id: int
    symbol: str
    side: str
    order_type: str
    quantity: float
    limit_price: float | None
    status: str
    mode: str
    submitted_at: datetime | None = None


class ExecutionResponse(BaseModel):
    id: int
    order_id: int
    executed_at: datetime
    price: float
    quantity: float
    fees: float | None = None


class PositionResponse(BaseModel):
    id: int
    symbol: str
    status: str
    quantity: float
    average_price: float | None = None
    market_value: float | None = None
    unrealized_pnl: float | None = None
    realized_pnl: float | None = None
    opened_at: datetime | None = None
